### PR TITLE
WIP - upgrade to Symfony 3

### PR DESCRIPTION
**Pull request by mickaelandrieu**
Be careful, there is a BC on the way we declare the routes. Of course, you can manage BC compatible way to do it, but I'm not interested to invest time on boring stuff :cat:

### CODE REVIEW

File: Routing/RouteCollection.php [hunk 1]
```diff
@@ -351,14 +351,19 @@ private function addRoute($name, array $route)
         $this->rawRoutes[$name] = $route;
 
         $newRoute = new Route(
-                $route['pattern'],
-                $route['defaults'],
-                array_key_exists('requirements', $route) ? $route['requirements'] : []
+                $route['path'],
```
Comment (mickaelandrieu) [inline Routing/RouteCollection.php]: Maybe something like:

```php
array_key_exists('path', $route) ? $route['path] : $route['pattern'],
```

**Comment by crouillon** [comment]
Hi @mickaelandrieu , thanks for this contrib but I've already worked on it on my own fork :-(
I'll soon push it on a new branch, for now please feel free to check it on https://github.com/crouillon/BackBee/tree/2.8-compliant
cu

**Comment by mickaelandrieu** [comment]
Is the migration of Symfony 3.4 something you'd consider? I've started to remove a lot of deprecations.

**Comment by crouillon** [comment]
Yes of course, we currently remove all deprecations from 2.8 to be about 3-ready

**Comment by mickaelandrieu** [comment]
Ok, you can re-use some parts of my pull request: mainly le routing, and the use of Event Dispatcher I noticed you have already fixed the quotes around @service notation. Is it helping if I redo this pull request on your fork, on the 2.8-compliant branch?

**Comment by crouillon** [comment]
Please, be my guest ;-)

**Comment by mickaelandrieu** [comment]
I'm sorry, I'm running out of time :/

Closed

**Pull request closed without merging**